### PR TITLE
ref(typing): add typings for tsdb/redis

### PR DESCRIPTION
**Pull request by anonrig**
Adds the remaining types for TSDB base and redis implementations

### CODE REVIEW

File: src/sentry/tsdb/base.py [hunk 1]
```diff
@@ -12,6 +15,12 @@
 ONE_DAY = ONE_HOUR * 24
 
 
+@dataclass
+class IncrMultiOptions(TypedDict):
```
Comment (asottile-sentry) [inline src/sentry/tsdb/base.py:19]: `@dataclass` and `TypedDict` is probably not correct?

File: src/sentry/tsdb/redis.py [hunk 2]
```diff
@@ -41,10 +47,10 @@ class SuppressionWrapper:
     def __init__(self, wrapped):
         self.wrapped = wrapped
 
-    def __enter__(self):
+    def __enter__(self) -> Any:
         return self.wrapped.__enter__()
```
Comment (asottile-sentry) [inline src/sentry/tsdb/redis.py:51]: could probably make this typesafe with `wrapped: ContextManager[T]` and have `__enter__` return `T` and have this class extend `Generic[T]`

File: src/sentry/tsdb/base.py (hunk 1 above)
Comment (anonrig) [reply src/sentry/tsdb/base.py:19]: Yup, nice catch.

File: src/sentry/tsdb/redis.py (hunk 2 above)
Comment (anonrig) [reply src/sentry/tsdb/redis.py:51]: Thanks! This is what I was looking for.

File: src/sentry/tsdb/base.py [hunk 3]
```diff
@@ -12,6 +15,12 @@
 ONE_DAY = ONE_HOUR * 24
 
 
+@dataclasses.dataclass
+class IncrMultiOptions:
```
Comment (asottile-sentry) [inline src/sentry/tsdb/base.py:19]: is this right? I don't see anything constructing this

Comment (anonrig) [reply src/sentry/tsdb/base.py:19]: It's used as a type on `incr_multi` parameter

Comment (asottile-sentry) [reply src/sentry/tsdb/base.py:19]: yeah I see that part -- but this is a concrete class so something else would need to be constructing this and passing it in for that to ever be possible?

**Review by asottile-sentry — APPROVED** [review]

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**